csharp/test/Apache.Arrow.Tests/ArrowArrayConcatenatorTests.cs

```diff
@@ -604,10 +598,11 @@ public void Visit(UnionType type)
 
                     for (int j = 0; j < dataList.Count; j++)
                     {
-                        byte index = (byte)Math.Max(j % 3, 1);
+                        byte index = (byte)Math.Min(j % 3, 1);
```

Comment:
I'm pretty sure this was the intended behaviour, otherwise the `index == 0` branch below is never used.

Reply:
With the original code, the indexes would be 1, 1, 2, 1, 1, 2, etc.. With this change, the indexes would be 0, 1, 1, 0, 1, 1, etc.